driver-sync/src/main/com/mongodb/client/internal/CryptConnection.java

```diff
@@ -148,9 +150,9 @@ private FieldNameValidator getFieldNameValidator(final SplittablePayload payload
 
     private MessageSettings createSplittablePayloadMessageSettings() {
         return MessageSettings.builder()
-                .maxBatchCount(wrapped.getDescription().getMaxBatchCount())
-                .maxMessageSize(MAX_MESSAGE_SIZE)
-                .maxDocumentSize(MAX_DOCUMENT_SIZE)
+                .maxBatchCount(getDescription().getMaxBatchCount())
+                .maxMessageSize(getDescription().getMaxMessageSize())
+                .maxDocumentSize(getDescription().getMaxDocumentSize())
```

Comment:
Now we just use the normal max document size (16MB) to control the size of the command document as a whole, so that we can so a single write that is 16MB